src/os/bluestore/BlueStore.cc

```diff
@@ -3689,10 +3689,20 @@ void BlueStore::_set_blob_size()
 
 int BlueStore::_set_cache_sizes()
 {
+  assert(bdev);
+  if (cct->_conf->bluestore_cache_size) {
+    cache_size = cct->_conf->bluestore_cache_size;
+  } else {
+    // choose global cache size based on backend type
+    if (bdev->is_rotational()) {
+      cache_size = cct->_conf->bluestore_cache_size_hdd;
+    } else {
+      cache_size = cct->_conf->bluestore_cache_size_ssd;
+    }
+  }
   cache_meta_ratio = cct->_conf->bluestore_cache_meta_ratio;
   cache_kv_ratio = cct->_conf->bluestore_cache_kv_ratio;
 
-  double cache_size = cct->_conf->bluestore_cache_size;
```

Comment:
@liewegas Sorry, resolved a conflict with Mark's PR https://github.com/ceph/ceph/pull/15786 just now.
Also fix a typo introduced by that PR.
Do you mind reloading this into the testing branch?